src/spyglass/spikesorting/spikesorting_curation.py

```diff
@@ -533,13 +546,54 @@ def _get_peak_channel(waveform_extractor: si.WaveformExtractor, peak_sign: str,
     peak_channel = {key : int(val) for key, val in peak_channel_dict.items()}
     return peak_channel
 
+def _get_cosine_similarity(waveform_extractor: si.WaveformExtractor, **metric_params):
+    """Gets the cosine similarity as computed by SpikeInterface's compute_template_similarity
+    """
+    cos_sim_mat = si.postprocessing.compute_template_similarity(
+        waveform_extractor=waveform_extractor,
+        **metric_params)
+    unit_ids = waveform_extractor.sorting.unit_ids
+    cos_sim_dict = {
+        unit_id1: {unit_id2: np.float64(cos_sim_mat[i, j])
+        for j, unit_id2 in enumerate(unit_ids)}
+        for i, unit_id1 in enumerate(unit_ids)
+        }
+    return cos_sim_dict
+
+def _get_correlogram_asymmetry(waveform_extractor: si.WaveformExtractor, **metric_params):
+    """Computes the correlogram asymmetry between two units using spikeinterface compute_correlogram
+    """
+
+    corr, bins = si.postprocessing.compute_correlograms(
+        waveform_extractor, **metric_params)
+    # To only use left edge of bin
+    bins = bins[:-1]
+    neg_inds = np.where((bins < 0) & (bins>-20))[0]
+    pos_inds = np.where((bins > 0) & (bins<20))[0]
+    zero_inds = np.where(bins==0)[0]
+    # asym_mat = np.zeros(shape=(corr.shape[0], corr.shape[1]))
+    unit_ids = waveform_extractor.sorting.unit_ids
+    asym_dict = {unit_id : dict.fromkeys(unit_ids) for unit_id in unit_ids}
+    for i, unit_id1 in enumerate(unit_ids):
+        for j, unit_id2 in enumerate(unit_ids):
+            ccg = corr[i,j]
+            neg_count = np.sum(ccg[neg_inds])
+            pos_count = np.sum(ccg[pos_inds])
+            zero_count = np.sum(ccg[zero_inds])
+            asym_dict[unit_id1][unit_id2] = np.float64(
```

Comment:
* why are we including the spikes at zero to compute asymmetry?
* how about `(pos_count-neg_count) / (pos_count+neg_count)`? This index is bounded between -1 and 1 and conveys a degree of asymmetry along with its  direction (unlike the currently defined metric). 
* again, do we want to include asymmetry index between every pair of neurons?

Reply:
Yeah, this does seem to bias the asymmetry more by adding `zero_count` to whichever side is already larger. 
@rsnevers wrote this... Rhino, do you have thoughts on Kyu's suggestion?

I think we would have to calculate the entire dictionary before determining which are the 10 most asymmetric units, or however you were wanting to limit it.

Reply:
@jguides Do you have thoughts on Kyu's suggestion?

Reply:
Thanks @dpeg22 for looping me in. I wrote the correlogram asymmetry metric and am happy to comment on the above. I included spikes at zero in the metric because I think this is necessary to correctly quantify asymmetry. To illustrate this, consider a correlogram with all its density at the origin. This is perfectly symmetric, and the metric I wrote captures this. Conceptually, in the formula, we are giving half the counts at zero to one side of the origin and half to the other (note that we add half of the counts at zero in numerator, and divide by total counts at zero in the denominator-- so no bias is introduced here). As another example, consider a correlogram with almost all its density at zero, a single count to the left of zero, and no counts to the right of zero. In this case, the correlogram is nearly symmetric, and again to capture this we need to consider the counts at zero. If we ignore them (e.g. with the alternative metric proposed), in this example we would say the correlogram is perfectly asymmetric, which I think would be a mistake. 

In terms of the metric capturing only magnitude of asymmetry and not direction, this was a design choice. I opted for modularity here (magnitude and direction are distinct quantities, and one could conceivably be used without the other). However, I felt torn myself about this at the time. I can also see reasons for combining magnitude and direction of asymmetry in a single metric (it's less code and if people end up wanting to extract magnitude or direction alone they could write additional code to do so). So I could see going either way on this.

Reply:
@jguides I see. You're right that if all the density is at zero, then the metric I suggested will divide zero by zero, which will probably return an error. I think in that case we can just return zero though. 

I'm still not sure if we should be adding the density at zero. In your second example, if every density is at zero except for 1 to the left, why is it a mistake to say that it is perfectly asymmetric? Seems like it should be treated the same as if the density at zero is 0 and there is only one count to the left. Otherwise depending on the density at zero you get different values.

I think it's fine to not reflect the direction if you think that is easier to interpret. Taking the max is reasonable. But I think your metric may need to be rescaled because it cannot be smaller than 0.5. For example, let's say you have a 100 counts to the left and 100 counts to the right (and 0 at zero). Then you have perfect symmetry. But the metric is still 0.5 instead of being 0. Isn't it more intuitive to scale it such that it ranges from 0 to 1?

Reply:
Thanks @khl02007 for your thoughts. Let me see if I can explain why I think it's important that we don't throw away counts at zero. I believe the goal of the metric is to quantify the degree to which a unit does or does not tend to spike before or after another unit. One way that a unit can tend not to spike before or after another is if its spikes come both before and after the other unit's. These correspond to counts in the correlogram on either side of zero. The other way a unit can tend not to fire before or after another is if the two units fire at the same time. These correspond to counts in the correlogram at zero. If we throw away counts at zero, we miss a way that a unit can tend not to spike before or after another. The second example above is an illustration of this.

In terms of the scale of the metric, I agree that a scale of 0 to 1 is more intuitive than from 0.5 to 1. I would be fine if we rescaled the metric to this range. This would require multiplying the current metric by two and then subtracting one.

Reply:
I see. But isn't it highly unlikely that there will be two spikes from different neurons that co-occur? Also: I think your rationale justifies adding the count at zero to the denominator, but then why add half of it to the numerator?

Reply:
Yes, definitely highly unlikely. However, I think it would be good to have metric that accounts for all possible cases. You can think of this as adding half of the zero counts to the side with more density; the other half are given to the side with less density.

Reply:
I understand but why assign half of density at zero to e.g. the density left of zero?